Remove Navigation screen from experiments page

getdave opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

📣 This is not a formal removal or ending of the Navigation Editor project. It is a stop gap whilst we consider the future of the project. See `Why` below.

As suggested in https://github.com/WordPress/gutenberg/issues/40822#issuecomment-1119238535 by @talldan, this PR removes the Navigation screen from the Gutenberg Experiments page. This means it can now only be accessed directly via URL.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Contributors are getting bug reports regarding this screen but it is not currently being actively maintained or worked on. This takes up time.

I decided against completely deleting the feature because there's been no official discussion or decision on that. However, this is a nice simple step that will help stem the flow of Issues being reported for a project which is not maintained.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

Removes the screen from the PHP which registers items on the Experiments page.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

1. Goto Experiments page at http://localhost:8888/wp-admin/admin.php?page=gutenberg-experiments
2. Verify that the Navigation Editor experiment no longer appears.

## Screenshots or screencast <!-- if applicable -->

getdave committed:
Remove Navigation screen from experiments page

talldan commented:
Nice, thanks for doing that so quickly! 

There might be some users who have the option already enabled and will be unable to turn it off with this change. I think this is the code that adds the wp admin menu entry:
https://github.com/WordPress/gutenberg/blob/df8647c1c6bcaa102f3cc6cfe43ec795d0f3bb2b/lib/init.php#L33-L44

I wonder if it's worth doing something like this:
```php
        // The navigation editor menu item is temporarily removed by hardcoding `false` here.
        // See https://github.com/WordPress/gutenberg/pull/40878 for more info.
	if ( false && get_option( 'gutenberg-experiments' ) ) {
```

What do you think?

getdave commented:
Great thinking @talldan as always 🙇

draganescu reviewed (approved):
Aside from @talldan 's great suggestion this is good.

getdave committed:
Hide admin menu item even if option is active

Kudos to @talldan https://github.com/WordPress/gutenberg/pull/40878#issuecomment-1119389147

getdave commented:
OK @talldan I've made the changes. What do you think?

adamziel commented on lib/init.php:
```diff
@@ -30,7 +30,9 @@ function gutenberg_menu() {
-	if ( get_option( 'gutenberg-experiments' ) ) {
+	// The navigation editor menu item is temporarily removed by hardcoding `false` here.
+	// See https://github.com/WordPress/gutenberg/pull/40878 for more info.
+	if ( false && get_option( 'gutenberg-experiments' ) ) {
```
This introduces a dead code branch. How about removing lines 34-44 entirely?

draganescu replied:
I agree with @adamziel that we should remove this code as well since we removed the option to opt in, the menu item code is of no use, had we surrounded with a dead branch condition the `add_settings_field ` code then we could argue we may come back to it, but given that is removed this other bit should also be just removed.

gziolo replied:
@getdave, you can always go to this PR and use the "Revert" button to open another PR that brings back all the removed changes. There is no value in keeping dead code when we have so great tools around version control.

getdave replied:
That's just it though. We don't necessarily want to remove this code. All we're doing here is hiding the experiment until such time as we (may/may not) decide to remove the Navigation screen project.

I'm not convinced we're ready to do that. Neither is @talldan. That's why this was aiming to be a simple "hide the experiment" PR 😄

adamziel replied:
@getdave I agree! Notice we're suggesting removing these specific 10 lines of code responsible for rendering the link. We are not talking about removing the Navigation screen feature by and large.

getdave replied:
I agree that we can remove this _specific_ piece of code however. We don't need to have it sitting around and it can be rolled back.

What I'm saying is that the decision has not been made to remove the entire Navigation Editor so I won't be removing that in this (or any other) PR...yet.

adamziel replied:
> What I'm saying is that the decision has not been made to remove the entire Navigation Editor so I won't be removing that in this (or any other) PR...yet.

Sure! Let's retain the feature and let's let these ten lines go :-)

getdave replied:
Ok I've removed the dead code branch.

getdave committed:
Remove dead code entirely

getdave commented:
> @getdave, if the Navigation Screen has been removed, does that mean we could review and close pretty much all the open [related issues](https://github.com/WordPress/gutenberg/labels/%5BFeature%5D%20Navigation%20Screen)?

I think that's reasonable.
